Escaped quotation mark not handled properly

The following:

```rust
let foobar = "foo bar";
xshell::cmd!("ls \"{foobar}\"");
```

fails with:

```
error[E0765]: unterminated double quote string
  --> websites/print_css/src/main.rs:90:29
   |
90 | ...                   xshell::cmd!("ls \"{foobar}\"");
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Adding a space after the last escaped quotation mark works:

```rust
let foobar = "foo bar";
xshell::cmd!("ls \"{foobar}\" ");
```

--- a/tests/it/main.rs
+++ b/tests/it/main.rs
@@ -475,7 +475,7 @@ fn string_escapes() {
     setup();
 
     assert_eq!(cmd!("\"hello\"").to_string(), "\"hello\"");
-    assert_eq!(cmd!("\"\"\" asdf\" \"").to_string(), "\"\"\" asdf\" \"");
+    assert_eq!(cmd!("\"\"\"asdf\"\"\"").to_string(), r##""""asdf""""##);
     assert_eq!(cmd!("\\\\\\\\\\\\\\\\\\\\\\\\\\").to_string(), "\\\\\\\\\\\\\\\\\\\\\\\\\\");
 }
 

--- a/tests/it/main.rs
+++ b/tests/it/main.rs
@@ -470,6 +470,16 @@ fn formatting() {
     cmd!("cargo fmt --all -- --check").run().unwrap()
 }
 
+#[test]
+#[allow(unused_must_use)]
+fn string_escapes() {
+    setup();
+
+    cmd!("\"hello\"");
+    cmd!("\"\"\" asdf\" \"");
+    cmd!("\\\\\\\\\\\\\\\\\\\\\\\\\\");
+}
+
 fn sleep_ms(ms: u64) {
     thread::sleep(std::time::Duration::from_millis(ms))
 }

--- a/tests/it/main.rs
+++ b/tests/it/main.rs
@@ -471,13 +471,12 @@ fn formatting() {
 }
 
 #[test]
-#[allow(unused_must_use)]
 fn string_escapes() {
     setup();
 
-    cmd!("\"hello\"");
-    cmd!("\"\"\" asdf\" \"");
-    cmd!("\\\\\\\\\\\\\\\\\\\\\\\\\\");
+    assert_eq!(cmd!("\"hello\"").to_string(), "\"hello\"");
+    assert_eq!(cmd!("\"\"\" asdf\" \"").to_string(), "\"\"\" asdf\" \"");
+    assert_eq!(cmd!("\\\\\\\\\\\\\\\\\\\\\\\\\\").to_string(), "\\\\\\\\\\\\\\\\\\\\\\\\\\");
 }
 
 fn sleep_ms(ms: u64) {

--- a/xshell-macros/src/lib.rs
+++ b/xshell-macros/src/lib.rs
@@ -184,10 +184,6 @@ fn parse_ts(s: &str) -> TokenStream {
     s.parse().unwrap()
 }
 
-fn lit_ts(s: &str) -> TokenStream {
-    TokenTree::Literal(proc_macro::Literal::string(s)).into()
-}
-
 #[cfg(test)]
 mod tests {
     use super::*;

--- a/xshell-macros/src/lib.rs
+++ b/xshell-macros/src/lib.rs
@@ -106,8 +106,14 @@ fn shell_lex(
     })
 }
 
+/// Like trim_matches except only trims a maximum of 1 match
+fn strip_matches<'a>(s: &'a str, pattern: &str) -> &'a str {
+    s.strip_prefix(pattern).unwrap_or(s).strip_suffix(pattern).unwrap_or(s)
+}
+
 fn tokenize(cmd: &str) -> impl Iterator<Item = Result<Token<'_>>> + '_ {
-    let mut cmd = cmd.trim_matches('"');
+    let mut cmd = strip_matches(cmd, "\"");
+
     iter::from_fn(move || {
         let old_len = cmd.len();
         cmd = cmd.trim_start();
@@ -128,13 +134,13 @@ fn tokenize(cmd: &str) -> impl Iterator<Item = Result<Token<'_>>> + '_ {
     })
 }
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq, Eq)]
 struct Token<'a> {
     joined_to_prev: bool,
     text: &'a str,
     kind: TokenKind,
 }
-#[derive(Debug)]
+#[derive(Debug, PartialEq, Eq)]
 enum TokenKind {
     Word,
     String,
@@ -176,4 +182,22 @@ fn respan(ts: TokenStream, span: Span) -> TokenStream {
 
 fn parse_ts(s: &str) -> TokenStream {
     s.parse().unwrap()
+}
+
+fn lit_ts(s: &str) -> TokenStream {
+    TokenTree::Literal(proc_macro::Literal::string(s)).into()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use TokenKind::*;
+
+    #[test]
+    fn double_quoted() {
+        let res = tokenize(r#"""asdf"""#).collect::<Result<Vec<Token<'_>>>>().unwrap();
+        eprintln!("res = {:?}", res);
+
+        assert_eq!(&*res, [Token { joined_to_prev: true, text: "\"asdf\"", kind: Word }])
+    }
 }

--- a/xshell-macros/src/lib.rs
+++ b/xshell-macros/src/lib.rs
@@ -134,13 +134,13 @@ fn tokenize(cmd: &str) -> impl Iterator<Item = Result<Token<'_>>> + '_ {
     })
 }
 
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug)]
 struct Token<'a> {
     joined_to_prev: bool,
     text: &'a str,
     kind: TokenKind,
 }
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug)]
 enum TokenKind {
     Word,
     String,
@@ -182,18 +182,4 @@ fn respan(ts: TokenStream, span: Span) -> TokenStream {
 
 fn parse_ts(s: &str) -> TokenStream {
     s.parse().unwrap()
-}
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-    use TokenKind::*;
-
-    #[test]
-    fn double_quoted() {
-        let res = tokenize(r#"""asdf"""#).collect::<Result<Vec<Token<'_>>>>().unwrap();
-        eprintln!("res = {:?}", res);
-
-        assert_eq!(&*res, [Token { joined_to_prev: true, text: "\"asdf\"", kind: Word }])
-    }
 }

